core/src/mast/node/basic_block_node/mod.rs

```diff
@@ -88,6 +88,17 @@ impl BasicBlockNode {
         Self::with_decorators(operations, DecoratorList::new())
     }
 
+    /// Returns a new [`BasicBlockNode`] from values that are assumed to be correct.
+    /// Should only be used when the source of the inputs is trusted (e.g. deserialization).
+    pub fn new_unsafe(
```

Comment:
nit: to be consistent, I wonder if we should get rid of the `BasicBlockNode::new()` constructor and rename the current `BasicBlockNode::with_decorators()` constructor into `BasicBlockNode::new()` (I don't think the current `BasicBlockNode::new()` is used in too many places).

Reply:
Updated in this [commit](https://github.com/0xPolygonMiden/miden-vm/pull/1453/commits/fad447f56ff68cbcc5603849071ed99e85e01ad0).

It involved quite a few changes because I took the opportunity to address the TODO about the assert/error. The approach here is aligned with the higher level constructors we added a while ago with `Option<DecoratorList>`.